### PR TITLE
[AIRFLOW-4477] Add models methods to BQHook

### PR DESCRIPTION
**Pull request by ryanyuan**
Make sure you have checked _all_ steps below.

### Jira

- [x] My PR addresses the following [Airflow-4477](https://issues.apache.org/jira/browse/AIRFLOW-4477) issues and references them in the PR title. For example, "\[AIRFLOW-XXX\] My Airflow PR"

### Description

- [x] Here are some details about my PR, including screenshots of any UI changes:
Add v2.models to BigQueryHook
Reference: https://cloud.google.com/bigquery/docs/reference/rest/v2/models

### Tests

- [x] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:
test_bigquery_hook.py:TestBigQueryHookModelOperations.test_bigquery_list_model
test_bigquery_hook.py:TestBigQueryHookModelOperations.test_bigquery_get_model
test_bigquery_hook.py:TestBigQueryHookModelOperations.test_bigquery_patch_model
test_bigquery_hook.py:TestBigQueryHookModelOperations.test_bigquery_delete_model

### Commits

- [x] My commits all reference Jira issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation

- [x] In case of new functionality, my PR adds documentation that describes how to use it.
  - All the public functions and the classes in the PR contain docstrings that explain what it does
  - If you implement backwards incompatible changes, please leave a note in the [Updating.md](https://github.com/apache/airflow/blob/master/UPDATING.md) so we can assign it to a appropriate release

### Code Quality

- [x] Passes `flake8`

### CODE REVIEW

**Comment by ryanyuan** [comment]
PTAL @kaxil @potiuk

**Review by potiuk — CHANGES_REQUESTED** [review]
Are you planning to use the models in some Operator? I think you should add it together with the same PR

File: airflow/contrib/hooks/bigquery_hook.py [hunk 1]
```diff
@@ -1776,6 +1776,178 @@ def insert_all(self, project_id, dataset_id, table_id,
...
+        if model_id is None:
+            raise ValueError('model_id cannot be None.')
+
+        project_id = project_id if project_id else self.project_id
+
+        self.log.info('Retrieving BigQuery model %s from project: %s  Dataset:%s',
+                      model_id, project_id, dataset_id)
+
+        try:
+            model = self.service.models().get(
```
Comment (VFagundes) [inline airflow/contrib/hooks/bigquery_hook.py:1833]: how about instantiate the model outside of your `try/except`, to isolate your model data from your execution, then just leave the execute instruction inside of it?
something like:

```
    model = self.service.models().get( projectId=project_id,
                datasetId=dataset_id,
                modelId=model_id)
    try:
      big_query_model = model.execute(num_retiries=self.num_retries)
        self.log.info('BigQuery model retrieved successfully: In project %s '
                          'Dataset %s Model %s', project_id, dataset_id, model_id)
        return big_query_model
    except HttpError as err:
            raise AirflowException(
                'BigQuery models failed to retrieve. Error was: {}'.format(err.content)
            )
```

what do you think? :)

File: airflow/contrib/hooks/bigquery_hook.py [hunk 2]
```diff
@@ -1776,6 +1776,178 @@ def insert_all(self, project_id, dataset_id, table_id,
...
+            body['labels'] = labels
+
+        if expiration_time:
+            body['expirationTime'] = expiration_time
+
+        self.log.info('Patching BigQuery model %s from project: %s  Dataset:%s',
+                      model_id, project_id, dataset_id)
+
+        try:
+            model = self.service.models().patch(
```
Comment (VFagundes) [inline airflow/contrib/hooks/bigquery_hook.py:1904]: Same comment about it 
https://github.com/apache/airflow/pull/5268/commits#r284322025
:)

**Comment by mik-laj** [comment]
@ryanyuan Do you plan to continue working on this change?  I am organizing tickets in JIRA related to GCP  and I would like to know the current status of this change.

**Comment by ryanyuan** [comment]
@mik-laj I am not going to work on this until there are more endpoints exposed because so far we found we could run the query directly to achieve what the API can do.

**Comment by mik-laj** [comment]
Do you think it's worth closing the ticket now? Now it is a ticket without a proper use case.  it can introduce noise on Jira

**Comment by ryanyuan** [comment]
@mik-laj Agree. Closing this PR.

**Pull request closed without merging**